### PR TITLE
added centralized error handling, razorpay_webhook, and S3 integration; updated dynamic content serving and Socket.IO setup.

### PR DESCRIPTION
**Pull request by parthb26**
> Added **Centralized Error handling** by introducing Middleware.
> Added a **'/payment-webhook'** route to handle payment notifications.
> Configured routes to serve static files and handle dynamic content from **S3**, including downloading and rendering project files.
> Implemented functionality to serialize and **serve HTML content dynamically**.
> **Enhanced response headers** with correct content types.
> **Integrated Socket.IO** for real-time communication and onboarding event handling.

### CODE REVIEW

**Comment by parthb26** [comment]
Sure @ptmaroct

File: server.js [hunk 1]
```diff
@@ -268,30 +267,37 @@ app.use("/site/:siteId", async (req, res, next) => {
...
-
-const PORT = process.env.PORT || 5001;
+io.on("connection", (socket) => {
+  handleOnboardingSocketEvents(socket);
+});
 
+const PORT = process.env.PORT || 4000;
 server.listen(PORT, () => {
-  console.log(`Server running at ${PORT}`);
+  console.log(`Server is running on port ${PORT}`);
```
Comment (ptmaroct) [inline server.js:302]: interesting rephrase, can you explain the logic behind this?
Does the server run on port or at port?

Comment (parthb26) [reply server.js:302]: just thought to go with some of my terms, sorry for that, "at" will do too..

**Comment by parthb26** [comment]
> You can ignore the razorpay part and complete paypal webhook integration @parthb26 

@ptmaroct do you want to replace the razorpay part with paypal? Or should I keep both the webhooks??

**Pull request closed without merging**